src/modules/sensors/sensors.cpp

```diff
@@ -468,16 +477,34 @@ void Sensors::Run()
 			_voted_sensors_update.update_mag_comp_armed(_armed);
 		}
 
-		if (_actuator_ctrl_0_sub.updated()) {
-			actuator_controls_s controls {};
-			_actuator_ctrl_0_sub.copy(&controls);
-			_voted_sensors_update.update_mag_comp_throttle(controls.control[actuator_controls_s::INDEX_THROTTLE]);
+		//check mag power compensation type (change battery current subscription instance if necessary)
+		if ((MagCompensationType)_parameters.mag_comp_type == MagCompensationType::Current_inst0
+		    && _mag_comp_type != MagCompensationType::Current_inst0) {
+			_battery_status_sub = uORB::Subscription{ORB_ID(battery_status), 0};
 		}
 
-		if (_battery_status_sub.updated()) {
+		if ((MagCompensationType)_parameters.mag_comp_type == MagCompensationType::Current_inst1
+		    && _mag_comp_type != MagCompensationType::Current_inst1) {
+			_battery_status_sub = uORB::Subscription{ORB_ID(battery_status), 1};
```

Comment:
What if changed from instance 1 to 0?

Reply:
then the if-clause on line 481 should be true and the subscribe is changed to instance 0. Or did I miss something?

Reply:
Of course. I should really look at the code.